### PR TITLE
[BazelProfile] Support reading action count of Bazel profiles generated with Bazel 7

### PR DESCRIPTION
**Pull request by saraadams**
With Bazel 7, the action count events are no longer part of the `Main Thread`, as they no longer include a `tid`.
This PR changes the logic, so that the action count can be retrieved both for older styleand newer style profiles.

Contributes to #113.

### CODE REVIEW

**Comment by saraadams** [comment]
Highlighted in https://github.com/bazelbuild/bazel/issues/18548#issuecomment-1823361627
I plan to work on a patch to reintroduce the tid tomorrow, which would greatly simplify this PR. Some of the changes might still be beneficial, though.
Converting to draft for now.

**Comment by saraadams** [comment]
If https://github.com/bazelbuild/bazel/pull/20300 is accepted, the `tid` should again be present.

**Comment by saraadams** [comment]
(I also verified that for Bazel 6.4.0 the `tid` is still present.)

File: analyzer/java/com/engflow/bazel/invocation/analyzer/bazelprofile/ThreadId.java [hunk 1]
```diff
@@ -15,17 +15,18 @@
...
+  private final Integer threadId;
 
-  public ThreadId(int processId, int threadId) {
+  public ThreadId(int processId, @Nullable Integer threadId) {
     this.processId = processId;
     this.threadId = threadId;
   }
 
-  public int getThreadId() {
+  public Integer getThreadId() {
```
Comment (ulfjack) [inline analyzer/java/com/engflow/bazel/invocation/analyzer/bazelprofile/ThreadId.java:29]: Should this be marked `@Nullable`?

**Review by ulfjack — APPROVED** [review]

File: analyzer/java/com/engflow/bazel/invocation/analyzer/bazelprofile/ThreadId.java (hunk 1 above)
Comment (saraadams) [reply analyzer/java/com/engflow/bazel/invocation/analyzer/bazelprofile/ThreadId.java:29]: I am not planning to merge this PR (hence I converted it to a draft), given that the issue is hopefully resolved on the Bazel side (ensuring profiles do have `tid` for action count entries) - see https://github.com/bazelbuild/bazel/pull/20300.

**Comment by saraadams** [comment]
Closing, as https://github.com/bazelbuild/bazel/pull/20300 was merged and gets rid of the regression.

**Pull request closed without merging**